Dockerfile

```diff
@@ -1,19 +1,42 @@
 # syntax=docker/dockerfile:1
 
+ARG GO_VERSION=1.19.3
+ARG BASE_DEBIAN_DISTRO="bullseye"
+ARG GOLANG_IMAGE="golang:${GO_VERSION}-${BASE_DEBIAN_DISTRO}"
+ARG XX_VERSION=1.1.2
+
+ARG CONTAINERD_VERSION="v1.6.10"
+ARG RUNC_VERSION="v1.1.4"
+ARG TINI_VERSION="v0.19.0"
+ARG ROOTLESSKIT_VERSION="6222b477d4c3ce6eea2bcff0586e43c95d1c0bb7"
+ARG DOCKERCLI_CHANNEL="stable"
+ARG DOCKERCLI_VERSION="v17.06.2-ce"
+ARG VPNKIT_VERSION=0.5.0
+
 ARG CROSS="false"
 ARG SYSTEMD="false"
-ARG GO_VERSION=1.19.3
 ARG DEBIAN_FRONTEND=noninteractive
-ARG VPNKIT_VERSION=0.5.0
+ARG DOCKER_STATIC=1
```

Comment:
Needs static for integration tests on Jenkins (arm64): https://ci-next.docker.com/public/blue/organizations/jenkins/moby/detail/PR-44513/3/pipeline#step-154-log-526

```
[2022-11-23T14:59:56.268Z] --- FAIL: TestContinueAfterPluginCrash (62.99s)
[2022-11-23T14:59:56.268Z] FAIL
[2022-11-23T14:59:56.268Z] 
[2022-11-23T14:59:56.268Z] === Failed
[2022-11-23T14:59:56.268Z] === FAIL: arm64.integration.plugin.logging TestContinueAfterPluginCrash (62.99s)
[2022-11-23T14:59:56.268Z]     logging_linux_test.go:67: timeout waiting for container i/o
```

Reply:
Should this default to `0`, and only be set for CI? (need to think what's best). We have different make targets for static and non-static, correct?

Reply:
Atm installers build static binaries by default like:
* https://github.com/moby/moby/blob/85b80ce3bb68fe3fe78d70fcf1944ff598649e7f/hack/dockerfile/install/containerd.installer#L30-L31
* https://github.com/moby/moby/blob/85b80ce3bb68fe3fe78d70fcf1944ff598649e7f/hack/dockerfile/install/runc.installer#L21-L22
I think we should keep the same behavior in the meantime but can reconsider.